internal/agent/hooks/lifecycle.go

```diff
@@ -1,18 +1,22 @@
 package hook
 
 import (
-	"github.com/kairos-io/kairos-sdk/utils"
 	"github.com/kairos-io/kairos-agent/v2/pkg/config"
+	v1 "github.com/kairos-io/kairos-agent/v2/pkg/types/v1"
+	"github.com/kairos-io/kairos-sdk/utils"
+	"time"
 )
 
 type Lifecycle struct{}
 
-func (s Lifecycle) Run(c config.Config) error {
-	if c.Install.Reboot {
+func (s Lifecycle) Run(c config.Config, spec v1.Spec) error {
+	if spec.ShouldReboot() {
+		time.Sleep(5)
 		utils.Reboot()
 	}
 
-	if c.Install.Poweroff {
+	if spec.ShouldShutdown() {
+		time.Sleep(5)
```

Comment:
should we add a message, saying the action will take place in 5 seconds?

Reply:
I would but this would just print to stdout and I dont like that at all, because if you want to log to a file or make the install quiet then you get different outputs to different places  :/

I hope to soon(ish) merge the agent config+elemental config into just 1 main config and once that its done, because on the hook we have access to the config object we will be able to access teh config.Logger and log to it respecting the output, level, etc....

Reply:
yup agree, better wait for logger, maybe some TODO then? but up to you